### DISCUSSION TITLE
Go custom completions

### DISCUSSION THREAD

**Discussion by e2dk4r** [nushell/nushell#6362] [Q&A] [answered]
Hi everyone,

I was trying to implement custom completions for [Go] programming language.
[Go] is using `-` (0x2d) character for long flags.

## First Question
Is there any way to write long flags with `-` (0x2d) character in nushell?

Below sample nu file for testing.
You can use `use full/path/to/go-completion.nu *` to test
```
# go-completion.nu

# compile the packages named by the import paths, along with their dependencies
export extern "go build" [
  -race # enable data race detection.
]
```

this giver `parse_mismatch` error
```
Error: nu::parser::parse_mismatch (link)

  × Parse mismatch during operation.
   ╭─[path/to/go/go-completions.nu:4:1]
 4 │ export extern "go build" [
 5 │   -race # enable data race detection.
   ·   ──┬──
   ·     ╰── expected short flag
 6 │ ]
   ╰────
```

## Second Question
Is there a way to set flags as `PARAMETER=VALUE` format.
I was trying to use `go build --buildvcs='true' main.go` but it 
gives `named files must be .go files: true` error.

If I try to execute without nushell parsing
`^go build -buildvcs='true' main.go`
it compiles

Below sample nu file for testing.
You can use `use full/path/to/go-completion.nu *` to test
```
# go-completions.nu
def goBuildVcsOptions [] {
  [ 'auto', "'true'", "'false'" ]
}
export extern "go build" [
  --buildvcs: string@goBuildVcsOptions # Whether to stamp binaries with version control information
]
```
```
# main.go
package main

func main() {
}
```

I am using
```
os:      win10
arch:    amd64
nushell: 0.67.0
go:      1.19
```
[Go]: https://go.dev

**Comment by fdncred** [comment, marked as answer]
> Is there any way to write long flags with - (0x2d) character in nushell?

Not at the moment. Long parameters start with 2 dashes. However, we've talked about allowing long params with one dash.
